game-core/src/main/java/games/strategy/engine/framework/GameRunner.java

```diff
@@ -163,8 +159,8 @@ public static void main(final String[] args) throws InterruptedException {
       Application.launch(TripleA.class, args);
     } else {
       SwingUtilities.invokeLater(() -> {
-        setupPanelModel.showSelectType();
         mainFrame = newMainFrame();
+        setupPanelModel.showSelectType();
```

Comment:
The needed context is with this update: https://github.com/triplea-game/triplea/pull/3584/files#diff-80614d8aaef02b03e24f48595cf4436aL112

Before the ordering did not matter. With the previous ordering:
- we would set the panel on our model (firing a listener event with nobody listening)
- create main frame and find the panel, then self-fire a change event

If the ordering was reversed, the code was written to be able to handle that too (this is not actually a feature, it's dead code and added complexity, but... we could handle reversed line ordering which is a plus):
- we create main panel, it had a null panel check and would skip the self-fire of the change event
- we would set the panel on the model, this time with main panel listening it would be updated.

To reduce coupling, the 'self-fire' of the change event in MainPanel is removed. So what happens now is we create the main panel, we then set a setup panel in the model and that fires an event which will then trigger the initial screen to be drawn on main panel.

The trade-off for the reduced coupling and removed dead-code is now we have a line ordering dependency. Solving that is being left for another branch/day.